src/gensxg.c

```diff
@@ -0,0 +1,339 @@
+  sxg_buffer_resize(filesize, buf);
+  fread(buf->data, sizeof(char), filesize, file);
+  fclose(file);
+}
+
+static void Help() {
+  fprintf(stderr,
+        "-help\n"
+        "  Show this message.\n"
+        "-url string\n"
+        "  The URI of the resource represented in the SXG file. (required)\n"
+        "-certUrl string\n"
+        "  The URI of certificate cbor file published. (required)\n"
+        "-validityUrl string\n"
+        "  The URI of validity information provided. (required)\n"
+        "-certificate string\n"
+        "  The certificate PEM file for the SXG. (mutual exclusive with "
+        "-publicKey)\n"
+        "-publicKey string\n"
+        "  The Ed25519 PEM file for the SXG. (mutual exclusive with "
+        "-certificate)\n"
+        "-private_key string\n"
+        "  The private key PEM file for the SXG. (required)\n"
+        "-content string\n"
+        "  Source file to be used as SXG payload (default [index.html]).\n"
+        "-contentType string\n"
+        "  Mime type of Source file (default [text/html]).\n"
+        "-date int\n"
+        "  The datetime for the SXG in unixtime. Use now by default.\n"
+        "-expire int\n"
+        "  The expire time of the SXG in unixtime. (default <date> +7 days)\n"
+        "-miRecordSize int\n"
+        "  The record size of Merkle Integrity Content Encoding. "
+        "(default [4096])\n"
+          );
+}
+
+void Dump(const Options* opt) {
+  fprintf(stderr,
+          "url: %s\n"
+          "certUrl: %s\n"
+          "validityUrl: %s\n"
+          "certificate: %s\n"
+          "publicKey: %s\n"
+          "privateKey: %s\n"
+          "content: %s\n"
+          "contentType: %s\n"
+          "date: %ld\n"
+          "expire: %ld\n"
+          "miRecordSize: %ld\n",
+          opt->url,
+          opt->cert_url,
+          opt->validity_url,
+          opt->certificate,
+          opt->public_key,
+          opt->private_key,
+          opt->content,
+          opt->content_type,
+          opt->date,
+          opt->expire,
+          opt->mi_record_size);
+}
+
+int main(int argc, char* const argv[]) {
+  Options opt = ParseOptions(argc, argv);
+  if (opt.help) {
+    Help();
+    return 0;
+  }
+
+  if (!Validate(&opt)) {
+    Dump(&opt);
+    return 1;
+  }
+  
+  sxg_signer_list_t signers = sxg_empty_signer_list();;
+  LoadSigner(&opt, &signers);
+  
+  sxg_raw_response_t content = sxg_empty_raw_response();;
+  LoadContents(&opt, &content.payload);
+  if (!sxg_header_append_string("content-type", opt.content_type,
+                                &content.header)) {
+    fprintf(stderr, "Failed to append content-type header.\n");
+    return 1;
+  }
+  
+  sxg_encoded_response_t encoded = sxg_empty_encoded_response();
+  if (!sxg_encode_response(opt.mi_record_size, &content, &encoded)) {
+    fprintf(stderr, "Failed to encode contents.");
+    return 1;
+  }
+  
+  sxg_buffer_t result = sxg_empty_buffer();
+  if (!sxg_generate(opt.url, &signers,
```

Comment:
This line also fits in one line (should be shorter than Line 322).

Reply:
done.